src/meta/src/rpc/ddl_controller.rs

```diff
@@ -587,20 +633,235 @@ impl DdlController {
         Ok(())
     }
 
+    async fn check_cycle_for_sink(
```

Comment:
Can we do this in the frontend to make meta's code clearer?

Reply:
Yes, because the metadata on frontend could be stale.

Reply:
That's not a big deal. 😄 We have a version number in the table catalog. If the catalog is stale, it'll be rejected. In other words, there're still race cases if two DDLs are running in the meta service.